3.sc_Morphology_Linear_Modeling/scripts/1b.fit_linear_model_2beta.py

```diff
@@ -0,0 +1,150 @@
+import matplotlib.pyplot as plt
+
+pd.set_option("mode.chained_assignment", None)
+
+
+# In[2]:
+
+
+# Parameters
+celltype = "PBMC"
+
+
+# In[3]:
+
+
+# Define inputs
+feature_file = pathlib.Path(f"../data/{celltype}_preprocessed_sc_norm.parquet")
+feature_df = pq.read_table(feature_file).to_pandas()
+
+
+# In[4]:
+
+
+# if path does not exist, create one
+pathlib.Path(f"./results/{celltype}").mkdir(parents=True, exist_ok=True)
+
+# define output file path
+one_beta_output_file_path = pathlib.Path(f"./results/{celltype}/lm_one_beta.tsv")
+two_beta_output_file_path = pathlib.Path(f"./results/{celltype}/lm_two_beta.tsv")
+three_beta_output_file_path = pathlib.Path(f"./results/{celltype}/lm_three_beta.tsv")
+four_beta_output_file_path = pathlib.Path(f"./results/{celltype}/lm_four_beta.tsv")
+
+
+# In[5]:
+
+
+cp_features = infer_cp_features(feature_df)
+print(f"We are testing {len(cp_features)} features")
+
+new_line = "\n"
+print(
+    f"The unique Treatment-Dosages are: {f', {new_line}'.join((feature_df['twob_Metadata_Treatment_Dose_Inhibitor_Dose'].unique()))}"
+)
+
+
+# #### Complex Linear Modeling (cell count btea + 2 beta approach)
+# Here I run the same analysis as above but with dosage of a treatment being a factor in the linear model. All features and treatments will be exported into 1 file.
+#
+# Linear Model:
+# $y = \beta _{0}x+ \beta _{1}x+ \beta _{2}x+ \epsilon$ where;
+# $y$ is each feature
+# $x$ is the inputed variables
+# $\beta _{0}$ is the beta coefficient attributed to cell count.
+# $\beta _{1}$ is the beta coefficient attributed to Inducer, Inhibitor, and Inhibitor Dose.
+# $\beta _{2}$ is the beta coefficient attributed to Inducer dose.
+# $\epsilon$ is the residual variance not explained by factors in the model
+
+# In[6]:
+
+
+# Loop for each treatment then each feature
+
+# define the control and treatment
+# Setup linear modeling framework
+model_covariates = ["Metadata_number_of_singlecells"]
+control = "DMSO_DMSO_0.025__0.100"
+lm_results = []
+for treatment in feature_df["twob_Metadata_Treatment_Dose_Inhibitor_Dose"].unique():
+    dosage_treatments_list = [treatment, control]
+    print(dosage_treatments_list)
+    df = feature_df.query(
+        "twob_Metadata_Treatment_Dose_Inhibitor_Dose in @dosage_treatments_list"
+    )
+    # Add dummy matrix of categorical genotypes
+    df[["twob_Metadata_Treatment_Inhibitor_Dose", "Treatment_Dose"]] = df[
+        "twob_Metadata_Treatment_Dose_Inhibitor_Dose"
+    ].str.split("__", expand=True)
+    tmp_df = df.loc[
+        :,
+        (
+            "twob_Metadata_Treatment_Inhibitor_Dose",
+            "Treatment_Dose",
+        ),
+    ]
+
+    tmp_df["twob_Metadata_Treatment_Inhibitor_Dose"] = LabelEncoder().fit_transform(
+        tmp_df["twob_Metadata_Treatment_Inhibitor_Dose"]
+    )
+    tmp_df["Treatment_Dose"] = LabelEncoder().fit_transform(tmp_df["Treatment_Dose"])
+
+    X = pd.concat([df.loc[:, model_covariates], tmp_df], axis=1)
```

Comment:
Consider using a more descriptive variable than `X`

Reply:
After looking at the rest of the code it seems like you just wanted to match the parameter name so I think this is an okay name, but consider adding a comment above this line to explain what is happening/what X represents.